libcloud/compute/drivers/vcloud.py

```diff
@@ -1986,6 +1993,78 @@ def _change_vm_ipmode(self, vapp_or_vm_id, vm_ipmode):
             )
             self._wait_for_task_completion(res.object.get('href'))
 
+    def _update_or_insert_section(self, res, section, prev_section, text):
+        try:
+            res.object.find(
+                fixxpath(res.object, section)).text = text
+        except:
+            # "section" section does not exist, insert it just
+            # before "prev_section"
+            for i, e in enumerate(res.object):
+                tag = '{http://www.vmware.com/vcloud/v1.5}%s' % prev_section
+                if e.tag == tag:
+                    break
+            e = ET.Element(
+                '{http://www.vmware.com/vcloud/v1.5}%s' % section)
+            e.text = text
+            res.object.insert(i, e)
+        return res
+
+    def _change_vm_admin_password(self, vapp_or_vm_id, ex_admin_password):
```

Comment:
do you not think this might be a function that users might want to call publically? If so I would rename it ex_change_vm_admin_password